readthedocs/proxito/views/serve.py

```diff
@@ -37,38 +37,38 @@
 
 
 class ServePageRedirect(CDNCacheControlMixin, ServeRedirectMixin, ServeDocsMixin, View):
+    def get(self, request, subproject_slug=None, filename=""):
 
-    def get(self,
-            request,
-            project_slug=None,
-            subproject_slug=None,
-            version_slug=None,
-            filename='',
-    ):  # noqa
+        unresolved_domain = request.unresolved_domain
+        project = unresolved_domain.project
 
-        version_slug = self.get_version_from_host(request, version_slug)
-        final_project, lang_slug, version_slug, filename = _get_project_data_from_request(  # noqa
-            request,
-            project_slug=project_slug,
-            subproject_slug=subproject_slug,
-            lang_slug=None,
-            version_slug=version_slug,
-            filename=filename,
-        )
+        # Use the project from the domain, or use the subproject slug.
+        if subproject_slug:
+            project = get_object_or_404(
+                project.subprojects, alias=subproject_slug
+            ).child
 
-        if self._is_cache_enabled(final_project):
+        # Get the default version from the current project,
+        # or the version from the external domain.
+        if unresolved_domain.is_from_external_domain:
+            version_slug = unresolved_domain.external_version_slug
+        else:
+            version_slug = project.get_default_version()
```

Comment:
I'd like to eventually rename this method to `get_default_version_slug()`. It's always confusing to me.